### PR TITLE
Codenotary vcn 0.9.8 - System trust check

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

Add the latest version of vcn with a new endpoint handling. Add system trust check to validate the trust during runtime. Fix the issue that a content trust fails to take the system as unhealthy, also if we have not accepted the content and all is fine.

## Type of change

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/utils/codenotary.py [hunk 1]
```diff
@@ -90,6 +90,10 @@ async def vcn_validate(
 
     if _ATTR_ERROR in data_json:
         raise CodeNotaryBackendError(data_json[_ATTR_ERROR], _LOGGER.warning)
+    if proc.returncode != 0:
+        raise CodeNotaryError(
+            f"CodeNotary invalid exit code {proc.returncode}", _LOGGER.error
+        )
```
Comment (pvizeli) [inline supervisor/utils/codenotary.py:96]: That's wrong (self memo)

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -50,7 +52,8 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         network-manager \
         apparmor-utils \
         libpulse0 \
-    && bash <(curl https://getvcn.codenotary.com -L) \
+    && curl -Lo /bin/vcn https://github.com/codenotary/vcn/releases/download/${VCN_VERSION}/vcn-${VCN_VERSION}-linux-amd64-static \
```
Comment (ludeeus) [inline .devcontainer/Dockerfile:55]: We should use the same approach in the main `Dockerfile`

File: supervisor/resolution/checks/core_trust.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @property
+    def context(self) -> ContextType:
+        """Return a ContextType enum."""
+        return ContextType.CORE
+
+    @property
+    def states(self) -> List[CoreState]:
+        """Return a list of valid states when this check can run."""
+        return [CoreState.RUNNING]
```
Comment (ludeeus) [inline supervisor/resolution/checks/core_trust.py:59]:
```suggestion
        return [CoreState.RUNNING, CoreState.STARTUP]
```
We should check this on startup as well?

**Review by ludeeus — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (pvizeli) [reply .devcontainer/Dockerfile:55]: we can't because of all archs

**Pull request merged**